File: Python/programmers/kakao/level3/징검다리_건너기.py
def solution(stones, k):
    left, right = 1, int(2e8)
    while left <= right:
        tmp = stones[:]
        mid = (left+right)//2
        cnt = 0
        for t in tmp:
            if t - mid <= 0:
                cnt += 1
            else:
                cnt = 0
            if cnt >= k:
                break
        if cnt >= k:
            right = mid-1
        else:
            left = mid+1
    return left


stones = [2, 4, 5, 3, 2, 1, 4, 2, 5, 1]
k = 3
print(solution(stones, k)) # 3

File: Java/programmers/kakao/level3/징검다리건너기.java
package CodeTest.Java.programmers.kakao.level3;

public class 징검다리건너기 {
    public int solution(int[] stones, int k) {
        int left = 1, right = 200_000_000;

        while (left <= right) {
            int mid = (left+right)/2;
            int cnt = 0;
            for (int stone : stones) {
                if (stone <= mid) cnt++;
                else cnt = 0;
                if (cnt >= k) break;
            }
            if (cnt >= k) right = mid-1;
            else left = mid+1;
        }

        return left;
    }
}
